### PR TITLE
Python unstable 2021-11-25

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Python update of minor versions.

There seemed to be excessive x86_64-linux hydra capacity. So may as well.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/boto3/default.nix [hunk 1]
```diff
@@ -13,11 +13,11 @@
 
 buildPythonPackage rec {
   pname = "boto3";
-  version = "1.18.54"; # N.B: if you change this, change botocore and awscli to a matching version
+  version = "1.20.13"; # N.B: if you change this, change botocore and awscli to a matching version
```
Comment (thiagokokada) [inline pkgs/development/python-modules/boto3/default.nix:16]: Should we also update awscli :thinking: ?

File: pkgs/development/python-modules/frozendict/default.nix [hunk 2]
```diff
@@ -8,14 +8,14 @@
 
 buildPythonPackage rec {
   pname = "frozendict";
-  version = "2.0.7";  # 2.0.6 breaks canonicaljson
+  version = "2.1.0";  # 2.0.6 breaks canonicaljson
```
Comment (thiagokokada) [inline pkgs/development/python-modules/frozendict/default.nix:11]: Can we remove this comment :thinking: ?

**Review by thiagokokada — APPROVED** [review]

File: pkgs/development/python-modules/boto3/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/boto3/default.nix:16]: probably, I'll test that next

File: pkgs/development/python-modules/frozendict/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/frozendict/default.nix:11]: yea

**Comment by dotlambda** [comment]
Should we include https://github.com/NixOS/nixpkgs/pull/147524?

**Comment by dotlambda** [comment]
There's also https://github.com/NixOS/nixpkgs/pull/145807 but I don't think we want that just yet?

File: pkgs/development/python-modules/boto3/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/boto3/default.nix:16]: fixed

File: pkgs/development/python-modules/frozendict/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/frozendict/default.nix:11]: done

**Comment by jonringer** [comment]
@fabaff can you avoid force pushing to the branch when not necessary.

You deleted a few of my commits

**Comment by fabaff** [comment]
@jonringer, sorry, my bad.

**Comment by fabaff** [comment]
I went for `twentemilieu` and `libcst` too. Looks like this was the start of my questionable actions.

**Comment by fabaff** [comment]
I will fix the whitespace issue in ce26888 when I'm sure that nobody else is working.

**Comment by zowoq** [comment]
https://github.com/NixOS/nixpkgs/runs/4360946550?check_suite_focus=true

Sorry about the spurious output in addition to the correct whitespace error in the editorconfig check , it's since been reverted.

**Comment by jonringer** [comment]
@fabaff I fixed the extra whitespace

**Comment by dotlambda** [comment]
Webssh should be moved to `all-packages.nix`.

**Comment by jonringer** [comment]
waiting for ofborg to go green

**Comment by dotlambda** [comment]
So no #147524?

**Comment by jonringer** [comment]
1 comment

**Comment by jonringer** [comment]
@ofborg eval

**Comment by fabaff** [comment]
Looks like that `matplotlib` is missing [`packaging`](https://github.com/matplotlib/matplotlib/blob/main/setup.py#L320) which became a requirement in 3.5.0.

Do we want to fix that here or later?

**Comment by jonringer** [comment]
> Looks like that `matplotlib` is missing [`packaging`](https://github.com/matplotlib/matplotlib/blob/v3.5.0/setup.py) which became a requirement in 3.5.0.
> 
> Do we want to fix that here or later?

Here, doing the final touches before merging to staging

**Comment by trofi** [comment]
`staging` and `staging-next` fail to build `bpython`:

```
$ nix build -f. python3Packages.bpython -L
...
python3.9-bpython> unpacking source archive /nix/store/9d0qp9mpxsvlxs9wkqvmbsyb02vpx7ff-bpython-0.22.1.tar.gz
python3.9-bpython> source root is bpython-0.22.1
python3.9-bpython> setting SOURCE_DATE_EPOCH to timestamp 1636367222 of file bpython-0.22.1/setup.cfg
python3.9-bpython> patching sources
python3.9-bpython> applying patch /nix/store/vz4q2sd2g43ham91p2f5sal27yikks7v-clipboard-make-which-substitutable.patch
python3.9-bpython> can't find file to patch at input line 16
python3.9-bpython> Perhaps you used the wrong -p or --strip option?
```

Bisect landed on  c064af6e4a3786baa4ce1d294734cd23a90eceb0 `python3Packages.bpython: 0.21 -> 0.22.1`.

**Comment by trofi** [comment]
Does `python3Packages.grpcio-status` fetch for you? On `staging` and `staging-next` it fails for me as:

```
$ nix build -f. python3Packages.grpcio-status -L
grpcio-status> trying https://files.pythonhosted.org/packages/source/g/grpcio-status/grpcio-status-1.43.0.tar.gz
grpcio-status>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
grpcio-status>                                  Dload  Upload   Total   Spent    Left  Speed
grpcio-status>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
grpcio-status> 100  8686  100  8686    0     0  54978      0 --:--:-- --:--:-- --:--:-- 54978
error: hash mismatch in fixed-output derivation '/nix/store/3pajxzh43idqywpp65gqcs72i17mix0d-grpcio-status-1.43.0.tar.gz.drv':
         specified: sha256-JVM8TWXX1ROmOPDEuIsrZnAOO/Q+aZWlWjvRXsC3eQI=
            got:    sha256-IXWQBvNqf/v/GH1BkfQRjActiqn6aCOhGq14QqPGzNA=
error: 1 dependencies of derivation '/nix/store/nl5jvlc1wygk4bnpiqmdxwhyfn6icvkg-python3.9-grpcio-status-1.43.0.drv' failed to build
```

**Comment by jonringer** [comment]
> Does `python3Packages.grpcio-status` fetch for you? On `staging` and `staging-next` it fails for me as:
> 
> ```
> $ nix build -f. python3Packages.grpcio-status -L
> grpcio-status> trying https://files.pythonhosted.org/packages/source/g/grpcio-status/grpcio-status-1.43.0.tar.gz
> grpcio-status>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> grpcio-status>                                  Dload  Upload   Total   Spent    Left  Speed
> grpcio-status>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> grpcio-status> 100  8686  100  8686    0     0  54978      0 --:--:-- --:--:-- --:--:-- 54978
> error: hash mismatch in fixed-output derivation '/nix/store/3pajxzh43idqywpp65gqcs72i17mix0d-grpcio-status-1.43.0.tar.gz.drv':
>          specified: sha256-JVM8TWXX1ROmOPDEuIsrZnAOO/Q+aZWlWjvRXsC3eQI=
>             got:    sha256-IXWQBvNqf/v/GH1BkfQRjActiqn6aCOhGq14QqPGzNA=
> error: 1 dependencies of derivation '/nix/store/nl5jvlc1wygk4bnpiqmdxwhyfn6icvkg-python3.9-grpcio-status-1.43.0.drv' failed to build
> ```

it inherits from grpc, so unless there's an update for that packge as well, just updating the grpc package will fail

**Comment by jonringer** [comment]
#151666

**Pull request merged**